lock-api/src/main/java/com/palantir/lock/client/TransactionStarter.java

```diff
@@ -70,12 +71,17 @@ static TransactionStarter create(LockLeaseService lockLeaseService, LockWatchEve
     }
 
     StartIdentifiedAtlasDbTransactionResponse startIdentifiedAtlasDbTransaction() {
-        try {
-            return autobatcher.apply(null).get();
-        } catch (ExecutionException e) {
-            throw Throwables.throwUncheckedException(e.getCause());
-        } catch (Throwable t) {
-            throw Throwables.throwUncheckedException(t);
+        return AtlasFutures.getUnchecked(autobatcher.apply(null));
+    }
+
+    StartIdentifiedAtlasDbTransactionResponseBatch startIdentifiedAtlasDbTransactionsBatch(int count) {
+        try (StartIdentifiedAtlasDbTransactionResponseBatch.Builder batchBuilder =
+                new StartIdentifiedAtlasDbTransactionResponseBatch.Builder(response ->
+                        unlock(ImmutableSet.of(response.immutableTimestamp().getLock())))) {
+            List<Void> inputs = IntStream.range(0, count).mapToObj($ -> (Void) null).collect(Collectors.toList());
+            autobatcher.applyBatch(inputs).forEach(
+                    response -> batchBuilder.safeAddToBatch(() -> AtlasFutures.getUnchecked(response)));
+            return batchBuilder.build();
```

Comment:
So in this class we call #getStartTransactionResponses which is obviously already batched. But it's implementation can call timelock multiple times, without any cleanup of resources between calls: @jeremyk-91 does StartIdentifiedAtlasDbTransactionResponse not require any cleanup if we fail mid batch in this method?

Reply:
The purpose of the Batch class is that if this does fail midway, we will call unlock (hence the passing in of the closing callback on line 80), unless you are referring to something else?

Reply:
Cool, I added tracking for this, we'll fix this next.